inc/sitemaps/class-sitemap-image-parser.php

```diff
@@ -62,19 +48,22 @@ public function get_images( $post ) {
 
 		$images = array();
 
-		$content = $post->post_content;
-		$content = '<p><img src="' . $this->image_url( get_post_thumbnail_id( $post->ID ) ) . '" alt="' . $post->post_title . '" /></p>' . $content;
-
-		if ( preg_match_all( '`<img [^>]+>`', $content, $matches ) ) {
-
-			$images = $this->parse_matched_images( $matches[0], $post );
+		if ( ! is_object( $post ) ) {
```

Comment:
We are querying less data than "normal" post object contains. We also tweak it with some hacks to speed up permalink generation.

This is de-facto _not_ a `WP_Post` object at this point in code. At this time I am not considering rewriting calling code for the sake of improving this check alone.